File: 2020/01/Part2.py
from typing import List

def read_input(name="input.txt") -> List[int]:
    l = []
    with open(name) as f:
        for line in f:
            if line.strip() != "":
                l.append(int(line.strip()))
    return l

def brute_force(l: List[int]) -> int:

    for index1 in range(len(l)):
        for index2 in range(index1+1, len(l)):
            for index3 in range(index2+1, len(l)):
                if l[index1] + l[index2] + l[index3] == 2020:
                    return l[index1] * l[index2] * l[index3]
    return 0

def main():
    l = read_input()
    print(brute_force(l))

def test():

    l = read_input("input_test.txt")

    assert brute_force(l) == 241861950, "Brute force failed on test input"

    print("All tests passed")

if __name__ == "__main__":
    from sys import argv
    if len(argv) >= 2 and argv[1] == "test":
        test()
    else:
        main()

File: 2020/15/Part2.java
public class Part2 {

    private static void test() {

        Shared.myAssert(Shared.solve("0,3,6", 30000000) == 175594L, "Failed test input 0");
        Shared.myAssert(Shared.solve("1,3,2", 30000000) == 2578L, "Failed test input 1");
        Shared.myAssert(Shared.solve("2,1,3", 30000000) == 3544142L, "Failed test input 2");
        Shared.myAssert(Shared.solve("1,2,3", 30000000) == 261214L, "Failed test input 3");
        Shared.myAssert(Shared.solve("2,3,1", 30000000) == 6895259L, "Failed test input 4");
        Shared.myAssert(Shared.solve("3,2,1", 30000000) == 18L, "Failed test input 5");
        Shared.myAssert(Shared.solve("3,1,2", 30000000) == 362L, "Failed test input 6");

        System.out.println("All tests passed");
    }

    public static void main(String[] args) {
        if (args.length >= 1 && args[0].equals("test")) {
            test();
        } else {
            System.out.println(Shared.solve("10,16,6,0,1,17", 30000000));
        }
    }
}